openrtb_ext/request_wrapper.go

```diff
@@ -31,46 +31,51 @@ import (
 // NOTE: The RequestWrapper methods (particularly the ones calling (un)Marshal are not thread safe)
 type RequestWrapper struct {
 	*openrtb2.BidRequest
-	imp         []*ImpWrapper
-	impAccessed bool
-	userExt     *UserExt
-	deviceExt   *DeviceExt
-	requestExt  *RequestExt
-	appExt      *AppExt
-	regExt      *RegExt
-	siteExt     *SiteExt
-	sourceExt   *SourceExt
+	impWrappers         []*ImpWrapper
+	impWrappersAccessed bool
+	userExt             *UserExt
+	deviceExt           *DeviceExt
+	requestExt          *RequestExt
+	appExt              *AppExt
+	regExt              *RegExt
+	siteExt             *SiteExt
+	sourceExt           *SourceExt
 }
 
 const jsonEmptyObjectLength = 2
 
 // LenImp returns the number of impressions without causing the creation of ImpWrapper objects.
 func (rw *RequestWrapper) LenImp() int {
-	if rw.imp == nil {
-		return len(rw.Imp)
+	if rw.impWrappersAccessed {
+		return len(rw.impWrappers)
 	}
 
-	return len(rw.imp)
+	return len(rw.Imp)
 }
 
 func (rw *RequestWrapper) GetImp() []*ImpWrapper {
-	if rw.imp != nil {
-		return rw.imp
+	if rw.impWrappersAccessed {
```

Comment:
Improved imp wrapper tracking to benefit tests. No impact expected for prod since there will always be at lease one impression per request.